### PR TITLE
Fixes and cleanup

### PR DESCRIPTION
**Pull request by matzapata**
## Overview

Multiple fixes and cleanup
- Add loading states to tables.
- Fix borrow power to match the customizable maximum collateral ratio
- Fix routing between native and token for the weth vs eth case.
- Close modals on complete
- Cleanup on transaction factories
- Implement max amount appropriately with `MaxUint256`
- Do allowances with `MaxUint256`
- Rollback contracts and other constants to single e-mode category
- Disable `switch emode` if only one category available
- Remove constant provider and use `useAccount` for it. This forced me out of `useCache` but I think it's alright, we can let them ask for it if that's required.

## Demo

supply token -> https://testnet-explorer.gobob.xyz/tx/0xda713e7ebfddefb4d7ef3f02f6ffd8308198236d34ab5d1a58ebe6c072d7cbef
withdraw token -> https://testnet-explorer.gobob.xyz/tx/0x6082108a5d0d6baab78cb79af55ed5bd1ac624b850b3e736d37d4149400d3092
withdraw token max -> https://testnet-explorer.gobob.xyz/tx/0xe1df9be4dfb7d0caaf439edb51e88e47512fb8fd18cfe95a12ef3a8a1927c186

supply native -> https://testnet-explorer.gobob.xyz/tx/0x5419f12ebe9fd70f6c8a02ec6cb8fe2c0c0862e596e984c633108bc6da95d533
withdraw native -> https://testnet-explorer.gobob.xyz/tx/0xa829a0f9767b2c702a59b8ba6908b3c1303d33a90c9438279dc03a59ccb51c88
withdraw native max -> https://testnet-explorer.gobob.xyz/tx/0x8bfb3933b49b3913c6ce234699dfa7c9c0babe12f74c00dbd0de65075f48046a

borrow token -> https://testnet-explorer.gobob.xyz/tx/0xc85606399597ee22d352d669250654d59298c9bbfd31057cadefb70a8371ea06
repay token -> https://testnet-explorer.gobob.xyz/tx/0x217e76f485486ceb74cf598c3d197fc5863dd9160aebe2fb63103a384934fc26
repay token max -> https://testnet-explorer.gobob.xyz/tx/0xb9a15e559c09e0850f04511db8c1b5a3281726102947ae58db972155975e0ca5

borrow native ->
https://testnet-explorer.gobob.xyz/tx/0x913f8c6548805b9135cd5c669bdbc358c736cd157384416a9dbf68f829fa46ce
https://testnet-explorer.gobob.xyz/tx/0xe71671b7f2ca75ff0380d9abdccc86491ff89d922485c0b29fd1f5013be5253d
repay native ->
https://testnet-explorer.gobob.xyz/tx/0xfece9839a11e3c9965464475536902f9fa9e2e298d6e3bd56a0b47abe2185a88
repay native max -> https://testnet-explorer.gobob.xyz/tx/0xda45b2da2515434a332ec7c54ee62664c4717af9213202faaf8c3e9d63e653fe

### CODE REVIEW

**Comment by matzapata** [comment]
I'll implement https://github.com/wakeuplabs-io/sovryn-dapp/pull/31#discussion_r1741111323 here

**Review by njaume — APPROVED** [review]
Great!

File: apps/frontend/src/utils/aave/AaveRepayTransactionsFactory.ts [hunk 1]
```diff
@@ -92,13 +100,15 @@ export class AaveRepayTransactionsFactory {
...
     amount: BigNumber,
+    isEntireDebt: boolean,
     borrowRateMode: BorrowRateMode,
     opts?: TransactionFactoryOptions,
   ): Promise<Transaction[]> {
     const nativeAsset = await getAssetDataByAddress(
       constants.AddressZero,
       BOB_CHAIN_ID,
     );
+    amount = isEntireDebt ? amount.add('1000000') : amount;
```
Comment (FacuDeLorenzo) [inline apps/frontend/src/utils/aave/AaveRepayTransactionsFactory.ts:111]: Is it ok to have this value **'1000000'** hardcoded?

Comment (matzapata) [reply apps/frontend/src/utils/aave/AaveRepayTransactionsFactory.ts:111]: Not ideal, maybe we can do a constant, but yeah, there's no specific rule for it other than "A bit more than debt". https://docs.aave.com/developers/periphery-contracts/wethgateway#repayeth for the contract field I can send uint256 but for value I don't think so... Any ideas are welcomed ))

Comment (matzapata) [reply apps/frontend/src/utils/aave/AaveRepayTransactionsFactory.ts:111]: ok, I'll make note of this and maybe come back with smth in different pr

**Pull request merged**